x-pack/plugins/ml/public/application/jobs/new_job/pages/components/common/datafeed_preview_flyout/datafeed_preview.tsx

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, useState, useEffect, useMemo } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiTitle,
+  EuiSpacer,
+  EuiLoadingSpinner,
+  EuiButton,
+} from '@elastic/eui';
+
+import { CombinedJob } from '../../../../../../../../common/types/anomaly_detection_jobs';
+import { MLJobEditor } from '../../../../../jobs_list/components/ml_job_editor';
+import { mlJobService } from '../../../../../../services/job_service';
+import { ML_DATA_PREVIEW_COUNT } from '../../../../../../../../common/util/job_utils';
+
+export const DatafeedPreview: FC<{
+  combinedJob: CombinedJob | null;
+  heightOffset?: number;
+}> = ({ combinedJob, heightOffset = 0 }) => {
+  const editorHeight = useMemo(() => `${window.innerHeight - 230 - heightOffset}px`, [
```

Comment:
nit: not sure if we need account for when window height changes but that maybe out of scope for this PR

Reply:
would be better to use `EuiResizeObserver`

Reply:
but at least we should avoid using the magic numbers (`- 230`)

Reply:
i agree that using magic numbers is not great, i'm inclined to leave this code as is because it can be removed once we get a better EUI code editor component that doesn't require an absolute height.
Adding code to measure the height of the callout is overly complex for a warning that will rarely appear.